src/app/core/internal/impl/CircuitInternal.ts

```diff
@@ -303,12 +316,22 @@ export class CircuitInternal extends Observable<FastCircuitDiff> {
     }
 
     public setCameraProps(props: Partial<Schema.Camera>) {
+        const dx = (props.x ?? this.camera.x) - this.camera.x;
+        const dy = (props.y ?? this.camera.y) - this.camera.y;
+        const dz = (props.zoom ?? this.camera.zoom) - this.camera.zoom;
+
+        // No change, no need to emit event
+        if (dx === 0 && dy === 0 && dz === 0)
```

Comment:
iirc the positions can be floats. Do you want to instead check that each is below some small value? Or does the whole usage of the null coalescing operator ensure that won't happen?

Reply:
the latter, yeah